mesonbuild/dependencies/python.py

```diff
@@ -45,6 +45,11 @@ class PythonIntrospectionDict(TypedDict):
     _Base = object
 
 
+if sys.version_info >= (3, 8):
+    from functools import cached_property
+else:
+    cached_property = property
```

Comment:
We have a `lazy_property` in utils somewhere now. It's provides more or less what cached_property does (although it's simpler and expects the attribute to be write once)

Reply:
I replaced `cached_property` with `lazy_property`.